src/golang/lib/engine/authenticate.go

```diff
@@ -32,6 +34,8 @@ func AuthenticateLambdaConfig(ctx context.Context, authConf auth.Config) error {
 		return errors.Wrap(err, "Unable to parse configuration.")
 	}
 
+	errGroup := new(errgroup.Group)
```

Comment:
I think the canonical way to do it is to use `errgroup.WithContext(ctx)` or `WithCancel()` here. That way when the outside context gets cancelled the goroutines will too. In this case, I don't think our goroutines actually use the context so it might not be that useful. However, when used in combination with `errgroup.GoContext()` instead of `errgroup.Go()`, you can at least make sure that obvious errors in the first goroutine will fail all the goroutines immediately. This is especially useful if you're sleeping in the loop.